src/crypto/DeviceList.js

```diff
@@ -431,39 +570,47 @@ class DeviceListUpdateSerialiser {
         // non-null indicates that we have users queued for download.
         this._queuedQueryDeferred = null;
 
-        // sync token to be used for the next query: essentially the
-        // most recent one we know about
-        this._nextSyncToken = null;
+        this._devices = null; // the complete device list
+        this._updatedDevices = null; // device list updates we've fetched
```

Comment:
could this be a local var in `_doQueuedQueries` ?

Reply:
or would it be easier just to poke _devices rather than return the updates?

Reply:
The actual updating happens in `_processQueryResponseForUser` hence why it's not just a local variable. Bluntly updating the _devices object would also work, I was mostly just trying to avoid modifying the object passed in by reference.